### ISSUE TITLE
Public key injection

### ISSUE THREAD

**Issue by mykter** [glassechidna/actions2aws#5]
I think there is a potential attack on the integrity of the public key.

An attacker can call the API at any point, and specify any run/job/step combo for which to search the output for a public key. If they can get their own public key included in the output of any step of any job, they can then obtain creds for the victim's account by calling the API specifying where it should look.

Whilst this won't be exploitable against all workflows, according to [Project Zero's research](https://bugs.chromium.org/p/project-zero/issues/detail?id=2070):

> almost any project with somewhat complex Github actions is vulnerable to this bug class

It's not obvious to users that if you're using actions2aws then allowing attacker-controlled text to appear in the output of any job grants that attacker access to your AWS account.

One mitigation would be to make the job/step config part of the lambda deployment, so it's not under attacker control.

**Comment by aidansteele** [comment]
Really nice thinking and great reference to P0, that's absolutely relevant here. I'm thinking out loud here, please correct me where I might be confused/incorrect:

* The first line of protection is refusing to vend credentials to any CI jobs from forks as per my response to the other issue you raised: https://github.com/glassechidna/actions2aws/issues/4#issuecomment-896499935. 
* If an attacker has the ability to inject text into a run that passes the above first check, that means that they were able to change the workflow of a commit inside the origin - which is only possible if they can push to the origin, right?
* It could well be useful (and maybe sensible) to make the job/step config part of the Lambda deployment. 🤔 But given that these are checked _after_ the run's associated HEAD repo (first check), it's probably not as helpful? What do you think

So I _think_ we're in the clear here. But as I said in the other response, I think this is _also_ worth making explicit in the README.

**Comment by mykter** [comment]
> which is only possible if they can push to the origin, right

Not necessarily, the P0 post describes both workflows that operate on issues in a project and also on pull requests between two existing branches in the project (i.e. not from a fork). So no forks required.

**Comment by aidansteele** [comment]
Oh I must have misunderstood, I will have to re-read the post more carefully. Thank you!
